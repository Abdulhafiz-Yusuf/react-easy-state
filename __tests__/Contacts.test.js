import React from 'react'
import { mount } from 'enzyme'
import App from '../examples/contacts/src/App'

describe('Contacts App', () => {
  const app = mount(<App />)

  test('should add new contacts', async () => {
    expect(app).toMatchSnapshot('01. Initial state')

    const creator = app.find('.contact-creator')
    const nameField = creator.find('input[name="name"]')
    const emailField = creator.find('input[name="email"]')
    const createButton = creator.find('button')

    nameField.simulate('change', {
      target: { name: 'name', value: 'Test Contact' }
    })
    expect(app).toMatchSnapshot('02. Create Test Contact name')

    emailField.simulate('change', {
      target: { name: 'email', value: 'test.contact@gmail.com' }
    })
    expect(app).toMatchSnapshot('03. Create Test Contact email')

    createButton.simulate('click')
    expect(app).toMatchSnapshot('04. Add Test Contact')

    nameField.simulate('change', {
      target: { name: 'name', value: '' }
    })
    emailField.simulate('change', {
      target: { name: 'email', value: '' }
    })
    createButton.simulate('click')
    expect(app).toMatchSnapshot('05. Add Placeholder Contact')
  })

  test('should edit contact', async () => {
    let display,
      editor,
      editButton,
      cancelButton,
      saveButton,
      nameField,
      emailField

    display = app.find('.contact-display').at(0)
    editButton = display.find('.zmdi-edit')

    editButton.simulate('click')
    expect(app).toMatchSnapshot('06. Switch Test Contact to Edit Mode')

    editor = app.find('.contact-editor').at(0)
    nameField = editor.find('input[name="name"]')
    cancelButton = editor.find('.zmdi-close')

    nameField.simulate('change', {
      target: { name: 'name', value: 'Edited Test Contact' }
    })
    expect(app).toMatchSnapshot('07. Edit Test Contact name')

    cancelButton.simulate('click')
    expect(app).toMatchSnapshot('08. Cancel Test Contact edit')

    display = app.find('.contact-display').at(0)
    editButton = display.find('.zmdi-edit')

    editButton.simulate('click')
    expect(app).toMatchSnapshot('09. Switch Test Contact to edit Mode')

    editor = app.find('.contact-editor').at(0)
    emailField = editor.find('input[name="email"]')
    saveButton = editor.find('.zmdi-save')

    emailField.simulate('change', {
      target: { name: 'email', value: 'test.contact.edited@gmail.com' }
    })
    expect(app).toMatchSnapshot('10. Edit Test Contact email')

    saveButton.simulate('click')
    expect(app).toMatchSnapshot('11. Save Test Contact edit')
  })

  test('should delete contact', async () => {
    let deleteButton = app.find('.contact-display .zmdi-delete').at(1)

    deleteButton.simulate('click')
    expect(app).toMatchSnapshot('12. Delete Placeholder Contact')

    deleteButton = app.find('.contact-display .zmdi-delete').at(0)

    deleteButton.simulate('click')
    expect(app).toMatchSnapshot('13. Delete Test Contact')
  })
})
