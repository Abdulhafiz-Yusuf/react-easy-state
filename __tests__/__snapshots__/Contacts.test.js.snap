// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts App should add new contacts: 01. Initial state 1`] = `
<App>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <ContactCreator>
        <tr
          className="contact-creator"
        >
          <td>
            <input
              name="name"
              onChange={[Function]}
              placeholder="Contact name..."
              value=""
            />
          </td>
          <td>
            <input
              name="email"
              onChange={[Function]}
              placeholder="Contact email..."
              value=""
            />
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              Add Contact
            </button>
          </td>
        </tr>
      </ContactCreator>
    </tbody>
  </table>
</App>
`;

exports[`Contacts App should add new contacts: 02. Create Test Contact name 1`] = `
<App>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <ContactCreator>
        <tr
          className="contact-creator"
        >
          <td>
            <input
              name="name"
              onChange={[Function]}
              placeholder="Contact name..."
              value="Test Contact"
            />
          </td>
          <td>
            <input
              name="email"
              onChange={[Function]}
              placeholder="Contact email..."
              value=""
            />
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              Add Contact
            </button>
          </td>
        </tr>
      </ContactCreator>
    </tbody>
  </table>
</App>
`;

exports[`Contacts App should add new contacts: 03. Create Test Contact email 1`] = `
<App>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <ContactCreator>
        <tr
          className="contact-creator"
        >
          <td>
            <input
              name="name"
              onChange={[Function]}
              placeholder="Contact name..."
              value="Test Contact"
            />
          </td>
          <td>
            <input
              name="email"
              onChange={[Function]}
              placeholder="Contact email..."
              value="test.contact@gmail.com"
            />
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              Add Contact
            </button>
          </td>
        </tr>
      </ContactCreator>
    </tbody>
  </table>
</App>
`;

exports[`Contacts App should add new contacts: 04. Add Test Contact 1`] = `
<App>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <Contact
        contact={
          Object {
            "email": "test.contact@gmail.com",
            "name": "Test Contact",
          }
        }
        key="test.contact@gmail.com"
      >
        <tr
          className="contact-display"
        >
          <td>
            Test Contact
          </td>
          <td>
            test.contact@gmail.com
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-edit"
              />
            </button>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-delete"
              />
            </button>
          </td>
        </tr>
      </Contact>
      <ContactCreator>
        <tr
          className="contact-creator"
        >
          <td>
            <input
              name="name"
              onChange={[Function]}
              placeholder="Contact name..."
              value=""
            />
          </td>
          <td>
            <input
              name="email"
              onChange={[Function]}
              placeholder="Contact email..."
              value=""
            />
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              Add Contact
            </button>
          </td>
        </tr>
      </ContactCreator>
    </tbody>
  </table>
</App>
`;

exports[`Contacts App should add new contacts: 05. Add Placeholder Contact 1`] = `
<App>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <Contact
        contact={
          Object {
            "email": "test.contact@gmail.com",
            "name": "Test Contact",
          }
        }
        key="test.contact@gmail.com"
      >
        <tr
          className="contact-display"
        >
          <td>
            Test Contact
          </td>
          <td>
            test.contact@gmail.com
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-edit"
              />
            </button>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-delete"
              />
            </button>
          </td>
        </tr>
      </Contact>
      <Contact
        contact={
          Object {
            "email": "Placeholder",
            "name": "Placeholder",
          }
        }
        key="Placeholder"
      >
        <tr
          className="contact-display"
        >
          <td>
            Placeholder
          </td>
          <td>
            Placeholder
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-edit"
              />
            </button>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-delete"
              />
            </button>
          </td>
        </tr>
      </Contact>
      <ContactCreator>
        <tr
          className="contact-creator"
        >
          <td>
            <input
              name="name"
              onChange={[Function]}
              placeholder="Contact name..."
              value=""
            />
          </td>
          <td>
            <input
              name="email"
              onChange={[Function]}
              placeholder="Contact email..."
              value=""
            />
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              Add Contact
            </button>
          </td>
        </tr>
      </ContactCreator>
    </tbody>
  </table>
</App>
`;

exports[`Contacts App should delete contact: 12. Delete Placeholder Contact 1`] = `
<App>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <Contact
        contact={
          Object {
            "email": "test.contact.edited@gmail.com",
            "name": "Test Contact",
          }
        }
        key="test.contact.edited@gmail.com"
      >
        <tr
          className="contact-display"
        >
          <td>
            Test Contact
          </td>
          <td>
            test.contact.edited@gmail.com
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-edit"
              />
            </button>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-delete"
              />
            </button>
          </td>
        </tr>
      </Contact>
      <ContactCreator>
        <tr
          className="contact-creator"
        >
          <td>
            <input
              name="name"
              onChange={[Function]}
              placeholder="Contact name..."
              value=""
            />
          </td>
          <td>
            <input
              name="email"
              onChange={[Function]}
              placeholder="Contact email..."
              value=""
            />
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              Add Contact
            </button>
          </td>
        </tr>
      </ContactCreator>
    </tbody>
  </table>
</App>
`;

exports[`Contacts App should delete contact: 13. Delete Test Contact 1`] = `
<App>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <ContactCreator>
        <tr
          className="contact-creator"
        >
          <td>
            <input
              name="name"
              onChange={[Function]}
              placeholder="Contact name..."
              value=""
            />
          </td>
          <td>
            <input
              name="email"
              onChange={[Function]}
              placeholder="Contact email..."
              value=""
            />
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              Add Contact
            </button>
          </td>
        </tr>
      </ContactCreator>
    </tbody>
  </table>
</App>
`;

exports[`Contacts App should edit contact: 06. Switch Test Contact to Edit Mode 1`] = `
<App>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <Contact
        contact={
          Object {
            "email": "test.contact@gmail.com",
            "name": "Test Contact",
          }
        }
        key="test.contact@gmail.com"
      >
        <tr
          className="contact-editor"
        >
          <td>
            <input
              autoFocus={true}
              name="name"
              onChange={[Function]}
              value="Test Contact"
            />
          </td>
          <td>
            <input
              name="email"
              onChange={[Function]}
              value="test.contact@gmail.com"
            />
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-save"
              />
            </button>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-close"
              />
            </button>
          </td>
        </tr>
      </Contact>
      <Contact
        contact={
          Object {
            "email": "Placeholder",
            "name": "Placeholder",
          }
        }
        key="Placeholder"
      >
        <tr
          className="contact-display"
        >
          <td>
            Placeholder
          </td>
          <td>
            Placeholder
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-edit"
              />
            </button>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-delete"
              />
            </button>
          </td>
        </tr>
      </Contact>
      <ContactCreator>
        <tr
          className="contact-creator"
        >
          <td>
            <input
              name="name"
              onChange={[Function]}
              placeholder="Contact name..."
              value=""
            />
          </td>
          <td>
            <input
              name="email"
              onChange={[Function]}
              placeholder="Contact email..."
              value=""
            />
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              Add Contact
            </button>
          </td>
        </tr>
      </ContactCreator>
    </tbody>
  </table>
</App>
`;

exports[`Contacts App should edit contact: 07. Edit Test Contact name 1`] = `
<App>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <Contact
        contact={
          Object {
            "email": "test.contact@gmail.com",
            "name": "Test Contact",
          }
        }
        key="test.contact@gmail.com"
      >
        <tr
          className="contact-editor"
        >
          <td>
            <input
              autoFocus={true}
              name="name"
              onChange={[Function]}
              value="Edited Test Contact"
            />
          </td>
          <td>
            <input
              name="email"
              onChange={[Function]}
              value="test.contact@gmail.com"
            />
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-save"
              />
            </button>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-close"
              />
            </button>
          </td>
        </tr>
      </Contact>
      <Contact
        contact={
          Object {
            "email": "Placeholder",
            "name": "Placeholder",
          }
        }
        key="Placeholder"
      >
        <tr
          className="contact-display"
        >
          <td>
            Placeholder
          </td>
          <td>
            Placeholder
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-edit"
              />
            </button>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-delete"
              />
            </button>
          </td>
        </tr>
      </Contact>
      <ContactCreator>
        <tr
          className="contact-creator"
        >
          <td>
            <input
              name="name"
              onChange={[Function]}
              placeholder="Contact name..."
              value=""
            />
          </td>
          <td>
            <input
              name="email"
              onChange={[Function]}
              placeholder="Contact email..."
              value=""
            />
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              Add Contact
            </button>
          </td>
        </tr>
      </ContactCreator>
    </tbody>
  </table>
</App>
`;

exports[`Contacts App should edit contact: 08. Cancel Test Contact edit 1`] = `
<App>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <Contact
        contact={
          Object {
            "email": "test.contact@gmail.com",
            "name": "Test Contact",
          }
        }
        key="test.contact@gmail.com"
      >
        <tr
          className="contact-display"
        >
          <td>
            Test Contact
          </td>
          <td>
            test.contact@gmail.com
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-edit"
              />
            </button>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-delete"
              />
            </button>
          </td>
        </tr>
      </Contact>
      <Contact
        contact={
          Object {
            "email": "Placeholder",
            "name": "Placeholder",
          }
        }
        key="Placeholder"
      >
        <tr
          className="contact-display"
        >
          <td>
            Placeholder
          </td>
          <td>
            Placeholder
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-edit"
              />
            </button>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-delete"
              />
            </button>
          </td>
        </tr>
      </Contact>
      <ContactCreator>
        <tr
          className="contact-creator"
        >
          <td>
            <input
              name="name"
              onChange={[Function]}
              placeholder="Contact name..."
              value=""
            />
          </td>
          <td>
            <input
              name="email"
              onChange={[Function]}
              placeholder="Contact email..."
              value=""
            />
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              Add Contact
            </button>
          </td>
        </tr>
      </ContactCreator>
    </tbody>
  </table>
</App>
`;

exports[`Contacts App should edit contact: 09. Switch Test Contact to edit Mode 1`] = `
<App>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <Contact
        contact={
          Object {
            "email": "test.contact@gmail.com",
            "name": "Test Contact",
          }
        }
        key="test.contact@gmail.com"
      >
        <tr
          className="contact-editor"
        >
          <td>
            <input
              autoFocus={true}
              name="name"
              onChange={[Function]}
              value="Test Contact"
            />
          </td>
          <td>
            <input
              name="email"
              onChange={[Function]}
              value="test.contact@gmail.com"
            />
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-save"
              />
            </button>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-close"
              />
            </button>
          </td>
        </tr>
      </Contact>
      <Contact
        contact={
          Object {
            "email": "Placeholder",
            "name": "Placeholder",
          }
        }
        key="Placeholder"
      >
        <tr
          className="contact-display"
        >
          <td>
            Placeholder
          </td>
          <td>
            Placeholder
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-edit"
              />
            </button>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-delete"
              />
            </button>
          </td>
        </tr>
      </Contact>
      <ContactCreator>
        <tr
          className="contact-creator"
        >
          <td>
            <input
              name="name"
              onChange={[Function]}
              placeholder="Contact name..."
              value=""
            />
          </td>
          <td>
            <input
              name="email"
              onChange={[Function]}
              placeholder="Contact email..."
              value=""
            />
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              Add Contact
            </button>
          </td>
        </tr>
      </ContactCreator>
    </tbody>
  </table>
</App>
`;

exports[`Contacts App should edit contact: 10. Edit Test Contact email 1`] = `
<App>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <Contact
        contact={
          Object {
            "email": "test.contact@gmail.com",
            "name": "Test Contact",
          }
        }
        key="test.contact@gmail.com"
      >
        <tr
          className="contact-editor"
        >
          <td>
            <input
              autoFocus={true}
              name="name"
              onChange={[Function]}
              value="Test Contact"
            />
          </td>
          <td>
            <input
              name="email"
              onChange={[Function]}
              value="test.contact.edited@gmail.com"
            />
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-save"
              />
            </button>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-close"
              />
            </button>
          </td>
        </tr>
      </Contact>
      <Contact
        contact={
          Object {
            "email": "Placeholder",
            "name": "Placeholder",
          }
        }
        key="Placeholder"
      >
        <tr
          className="contact-display"
        >
          <td>
            Placeholder
          </td>
          <td>
            Placeholder
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-edit"
              />
            </button>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-delete"
              />
            </button>
          </td>
        </tr>
      </Contact>
      <ContactCreator>
        <tr
          className="contact-creator"
        >
          <td>
            <input
              name="name"
              onChange={[Function]}
              placeholder="Contact name..."
              value=""
            />
          </td>
          <td>
            <input
              name="email"
              onChange={[Function]}
              placeholder="Contact email..."
              value=""
            />
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              Add Contact
            </button>
          </td>
        </tr>
      </ContactCreator>
    </tbody>
  </table>
</App>
`;

exports[`Contacts App should edit contact: 11. Save Test Contact edit 1`] = `
<App>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <Contact
        contact={
          Object {
            "email": "test.contact.edited@gmail.com",
            "name": "Test Contact",
          }
        }
        key="test.contact.edited@gmail.com"
      >
        <tr
          className="contact-display"
        >
          <td>
            Test Contact
          </td>
          <td>
            test.contact.edited@gmail.com
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-edit"
              />
            </button>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-delete"
              />
            </button>
          </td>
        </tr>
      </Contact>
      <Contact
        contact={
          Object {
            "email": "Placeholder",
            "name": "Placeholder",
          }
        }
        key="Placeholder"
      >
        <tr
          className="contact-display"
        >
          <td>
            Placeholder
          </td>
          <td>
            Placeholder
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-edit"
              />
            </button>
            <button
              onClick={[Function]}
            >
              <i
                className="zmdi zmdi-delete"
              />
            </button>
          </td>
        </tr>
      </Contact>
      <ContactCreator>
        <tr
          className="contact-creator"
        >
          <td>
            <input
              name="name"
              onChange={[Function]}
              placeholder="Contact name..."
              value=""
            />
          </td>
          <td>
            <input
              name="email"
              onChange={[Function]}
              placeholder="Contact email..."
              value=""
            />
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              Add Contact
            </button>
          </td>
        </tr>
      </ContactCreator>
    </tbody>
  </table>
</App>
`;
